ide/session.ml

```diff
@@ -171,14 +177,18 @@ let set_buffer_handlers
     let text_mark = add_mark min_iter in
     let rec aux min_iter =
       if min_iter#equal max_iter then ()
+      else if min_iter#has_tag Tags.Script.incomplete then
+        cancel_signal "Altering the script being processed in not implemented"
       else if min_iter#has_tag Tags.Script.to_process then
         cancel_signal "Altering the script being processed in not implemented"
       else if min_iter#has_tag Tags.Script.processed then
         call_coq_or_cancel_action (coqops#go_to_mark (`MARK text_mark))
       else if min_iter#has_tag Tags.Script.error_bg then
-        let prev_sentence_end = backto_before_error min_iter in
-        let text_mark = add_mark prev_sentence_end in
-        call_coq_or_cancel_action (coqops#go_to_mark (`MARK text_mark))
+        match processed_sentence_just_before_error min_iter with
```

Comment:
Can't we factorize all this logic between insert and delete?

Reply:
you are welcome ;-)